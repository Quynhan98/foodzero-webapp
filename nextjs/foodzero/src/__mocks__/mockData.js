export const AVATAR_MOCK =
  'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf'

export const PRICE_LIST_MOCK = {
  name: 'Deep Sea Snow White Cod Fillet',
  price: 20,
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ',
}

export const CARD_BLOG = {
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FFeature%20Image.webp?alt=media&token=766a34ea-32e0-40e7-90e6-299f9af22eb1',
  title: 'Fruit and vegetables and protection against diseases',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  category: 'Fashion',
  srcAvatar: AVATAR_MOCK,
  name: 'Julie Christie',
  date: '2022-12-06T23:29:35.000Z',
  numberOfComments: 2,
}

export const CARD_FEATURE = {
  title: 'Premium Quality',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2FCircleIcon.webp?alt=media&token=1df373f3-27a3-45f6-82fa-d997c47c718f',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Neque congue arcu',
}

export const CARD_CATEGORY = {
  href: '/',
  category: 'Starters',
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcategory.webp?alt=media&token=9f307d7c-b242-479d-943a-4ae672ac601e',
}

export const USERS_MOCK = [
  {
    id: 'gff1',
    email: 'foodzero@gmail.com',
    password: '123456789',
  },
]

export const QUOTE_MOCK = [
  {
    id: '1',
    authorName: 'John Doe',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fjohn-doe.webp?alt=media&token=6963f744-1ab8-4b09-971d-2486cd5375bf',
  },
  {
    id: '2',
    authorName: 'Julie Christie',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet1.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Funnamed.webp?alt=media&token=72862fa3-6567-4043-aab0-cc449b30e602',
  },
  {
    id: '3',
    authorName: 'Dianne Russell',
    job: 'Blogger',
    quote:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus impedient. Turpis egestas ultricies purus  Lorem ipsum dolor sit amet2.',
    avatar:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/foodzero%2Fcute-avatar.webp?alt=media&token=7b6d918e-3dec-43cf-a4bd-70afdbdc63e0',
  },
]

export const OUR_MENU_MOCK = [
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
  PRICE_LIST_MOCK,
]

export const FEATURES_MOCK = [CARD_FEATURE, CARD_FEATURE, CARD_FEATURE]

export const BLOG_SECTION_MOCK = [CARD_BLOG, CARD_BLOG]

export const CATEGORY_SECTION_MOCK = [
  CARD_CATEGORY,
  CARD_CATEGORY,
  CARD_CATEGORY,
]
