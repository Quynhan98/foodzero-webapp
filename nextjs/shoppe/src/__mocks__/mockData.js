export const PRODUCT_ITEM = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: 'Sold out',
}
