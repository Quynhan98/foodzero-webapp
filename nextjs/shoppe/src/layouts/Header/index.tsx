import Link from 'next/link'
import Image, { ImageLoader } from 'next/image'
import { Flex, Heading } from '@chakra-ui/react'

// Components
import Icon from '@components/Icon'

// Services
import myImageLoader from '@services/imageLoader'

const Header = () => {
  return (
    <Flex
      as="header"
      borderBottom="1px"
      borderBottomColor="lightSilver"
      justifyContent="space-between"
      alignItems="center"
      paddingBottom="5px"
      maxWidth="1248px"
      margin="0 auto"
      pt="67px"
    >
      <Heading as="h1">
        <Link href="/">
          <Image
            width={125}
            height={40}
            src="/images/logo.png"
            alt="Logo Shoppe"
            loader={myImageLoader as ImageLoader}
          />
        </Link>
      </Heading>
      <Icon
        width="21px"
        height="20px"
        marginRight="60px"
        srcIcon="/images/shoppingCart.svg"
        alt="Shopping Cart Icon"
      />
    </Flex>
  )
}

export default Header
