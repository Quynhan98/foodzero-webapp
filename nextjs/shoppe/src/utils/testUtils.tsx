import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>{component}</ChakraProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
