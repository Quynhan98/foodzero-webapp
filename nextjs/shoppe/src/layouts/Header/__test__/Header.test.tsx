// Utils
import { render } from '@utils/testUtils'

// Components
import Header from '@layouts/Header'

// Mocks
import { CARTS_MOCK } from '@mocks/mockData'

describe('Header render', () => {
  it('Should Header match Snapshot', () => {
    const { container } = render(<Header carts={CARTS_MOCK} />)

    expect(container).toMatchSnapshot()
  })
})
