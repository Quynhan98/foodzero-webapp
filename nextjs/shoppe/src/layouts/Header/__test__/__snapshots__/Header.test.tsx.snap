// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render Should Header match Snapshot 1`] = `
<div>
  <header
    class="css-13ughem"
  >
    <h1
      class="chakra-heading css-1dklj6k"
    >
      <img
        alt="Logo Shoppe"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=75"
        srcset="/_next/image?url=%2Fimages%2Flogo.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=256&q=75 2x"
        style="color: transparent;"
        width="125"
      />
    </h1>
    <div
      class="css-1bdr409"
    >
      <img
        alt="Shopping Cart Icon"
        class="chakra-image css-0"
        data-testid="image"
        src="/images/shoppingCart.svg"
      />
    </div>
  </header>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
